test/internal/Compose.js

```diff
@@ -23,11 +25,11 @@ module.exports = function Compose(F) {
     };
 
     ComposeFG.prototype[FL.map] = function(f) {
-      return ComposeFG(Z.map(function(y) { return Z.map(f, y); }, this.value));
+      return ComposeFG(Z.map(S.map(f), this.value));
     };
 
     ComposeFG.prototype[FL.ap] = function(other) {
-      return ComposeFG(Z.ap(Z.map(function(u) { return function(y) { return Z.ap(u, y); }; }, other.value), this.value));
+      return ComposeFG(Z.ap(Z.map(S.ap, other.value), this.value));
```

Comment:
That's right. Perhaps it would be better to use our internal [`ap`][1] function instead (in both cases). What do you think?


[1]: https://github.com/sanctuary-js/sanctuary/blob/0a704a0b72f806b4a3acd1306b7e1de899e82af2/test/internal/laws/ap.js